lib/util/git.js

```diff
@@ -141,8 +142,8 @@ function revs (repo, opts) {
           }
 
           if (type === 'tag') {
-            const match = ref.match(/v?(\d+\.\d+\.\d+)$/)
-            if (match) {
+            const match = ref.match(/v?(\d+\.\d+\.\d+(?:[-+].+)?)$/)
+            if (match && semver.valid(match[1])) {
```

Comment:
This needs to make sure it's using loose mode, so two changes:

1. `semver.valid()` needs to get `true` as its second arg, so it'll do validity checks in loose mode.
2. `semver.clean()` should be called on `match[1]` to clean up the spec. It also needs the second `true` arg for loose mode.

Reply:
Great, this will fix a big pain point for me in NPM. I just pushed a commit re your comments.